pkg/plugins/policies/meshtimeout/plugin/xds/configurer.go

```diff
@@ -94,19 +95,33 @@ func (c *DeprecatedListenerConfigurer) configureRequestTimeout(routeConfiguratio
 }
 
 func (c *DeprecatedListenerConfigurer) configureRequestHeadersTimeout(hcm *envoy_hcm.HttpConnectionManager) {
-	if conf := c.getConf(c.Subset); conf != nil {
+	// For backward, once a user upgrades from an older version we shouldn't set default timeouts.
+	// Refer to https://github.com/kumahq/kuma/issues/12033
+	deprecatedGetConf := c.legacyGetConf(c.Subset)
```

Comment:
What I don't understand with this code is when do we actually get rid of this? This code explains why there's so deprecated code but there should be a moment where we can remove this code right? When? Where's the issue tracking this?